client/components/track-clicks/index.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * External dependencies
+ */
+import storeShape from 'react-redux/lib/utils/storeShape';
+import { cloneElement, Children, Component, PropTypes } from 'react';
+import { constant, isFunction, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import pick from 'lib/pick';
+import { GUIDED_TOUR_TRACKED_CLICK } from 'state/action-types';
```

Comment:
Can we use a more generic action type since it's not GuidedTours-specific? Naming is hard, I agree. 😄 A few ideas:

- `COMPONENT_CLICK_TRACK`
- `TRACKED_COMPONENT_CLICK`
- `OBSERVED_COMPONENT_CLICK`

FWIW we could also need to discuss event types other than `click`, see below.